apps/ejabberd/src/mod_push_service_mongoosepush.erl

```diff
@@ -44,7 +44,8 @@
 %%--------------------------------------------------------------------
 
 -spec start(Host :: ejabberd:server(), Opts :: list()) -> any().
-start(Host, Opts) ->
+start(_ServerHost, Opts) ->
+    Host = ?MYNAME,
```

Comment:
This change may lead to module crashes when the server is configured with several virtual-hosts, because the module will start multiple times with the same `Host`. Why this is an issue for you? Sure, the current code will start multiple VHosts and the pubsub will use only one of them, but "it should work". The correct fix here would be not to use `?MYNAME ` in `node_push`. I don't really remember why I did that, but it may be because it was not trivial to get the host in that context.

Anyway, if you let me know what the real issue is, we may find better solution to this problem.